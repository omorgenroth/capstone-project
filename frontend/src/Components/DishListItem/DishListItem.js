import { FaMinusCircle } from 'react-icons/fa'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

ListItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  onDelete: PropTypes.func,
}

export default function ListItem({ id, title, onDelete }) {
  return (
    <>
      <ItemWrapper>
        <Title>{title}</Title>
        <DeleteIcon onClick={() => onDelete(id)} />
      </ItemWrapper>
      <Ruler />
    </>
  )
}

const ItemWrapper = styled.div`
  display: grid;
  grid-template-columns: 4fr 1fr;
`

const Title = styled.div`
  font-size: 1rem;
  grid-column: 1/2;
`

const DeleteIcon = styled(FaMinusCircle)`
  color: var(--c-orange);
  transform: scale(1);
  grid-column: 2/3;
`

const Ruler = styled.hr`
  border: 0.2px solid var(--c-green);
  width: 100%;
`
