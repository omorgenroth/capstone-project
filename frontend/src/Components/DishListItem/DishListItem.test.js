import { render } from '@testing-library/react'
import DishListItem from './DishListItem'

describe('DishListItem', () => {
  it('renders correctly', () => {
    const { container } = render(<DishListItem title="My Dish" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
