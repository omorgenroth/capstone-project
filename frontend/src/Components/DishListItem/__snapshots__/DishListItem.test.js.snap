// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishListItem renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 4fr 1fr;
}

.c1 {
  font-size: 1rem;
  grid-column: 1/2;
}

.c2 {
  color: var(--c-orange);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  grid-column: 2/3;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    My Dish
  </div>
  <svg
    class="c2"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
    />
  </svg>
</div>
`;
